web/modules/custom/brms_sor/src/Service/ForestReserveService.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\brms_sor\Service;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Service for retrieving Forest Reserves data.
+ */
+class ForestReserveService {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new ForestReserveService object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Get the master polygon for a forest reserve.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The forest reserve node.
+   *
+   * @return string|null
+   *   The master polygon.
+   */
+  public function getMasterPolygon(NodeInterface $node) {
+    if ($node->getType() != 'forest_reserve') {
+      return NULL;
+    }
+
+    $geolayers = $node->get('geolayers')->getValue();
+    foreach ($geolayers as $geolayer) {
+      $geolayerEntity = $this->entityTypeManager->getStorage('geolayer')->load($geolayer['target_id']);
+      $layerType = $geolayerEntity->get('layer_type')->entity;
+
+      if ($layerType && $layerType->getName() == 'Master polygon') {
```

Comment:
This is a bit convoluted, why are we comparing against a label?

Reply:
Feels that all this could be an entityQuery instead

Reply:
I've improved the logic for determining if the geolayer is the master polygon, by adding a master polygon flag to the layer type term. Now instead of comparing with the term name we look at the master_polygon flag.